Recipe card component

PeterBrandal opened the pull request:
See commit -m and discord messages.

PeterBrandal committed:
added recipe-detail page, altered recipe-card and app so all recipecards clicked will lead to the detailed recipe card

PeterBrandal committed:
now i have done the previous

PeterBrandal committed:
Favorites/Saved Recipes
Fixes #11

PeterBrandal committed:
added checkboxes, more components to details: profile pc, serv size, nutrients, bigger hero, rating

PeterBrandal committed:
update

PeterBrandal committed:
made responsive dashboard for screensizes of ipad size and upwards

PeterBrandal committed:
removed the data-folder which was caused by hallucinating AI

PeterBrandal committed:
new design for navbar drawer, also smoothened UX/UI for the mobile searchbar

PeterBrandal committed:
cook mode

PeterBrandal committed:
format

PeterBrandal committed:
added more attributes on recipe.ts / made it so recipe-detail will use fetched recipe data, if non-existent it will use the mock-data defined in recipe-detail

PeterBrandal committed:
better readability on recipe-detail

PeterBrandal committed:
fixed renderingdangers by moving all hooks before rendering

PeterBrandal committed:
better logic, removed unused imports, used recipe-loader instead, added filter-sidebar

PeterBrandal committed:
added link back to home on navbar

PeterBrandal committed:
Fixed duplicates of recipes existing in multiple recipe categories, now only renders one example of a dish that yet exists under many categoreies, such as ROSOL chicken soup, which can be found in both chicken and soup.JSON

PeterBrandal committed:
format

PeterBrandal committed:
created header for shorter code and beter modularity

kristtst01 commented on src/components/filter-panel.tsx:
Border overflows when filter bar is selected:
<img width="634" height="421" alt="image" src="https://github.com/user-attachments/assets/9e779920-f8e6-46cd-a2fd-b1f19a56de96" />
This won't work when we have every nation on earth:
<img width="683" height="779" alt="image" src="https://github.com/user-attachments/assets/d98fec93-89f7-4309-be6f-e820d21b872a" />
We can't have a dropdown like that. Should revert to the way it was with search.

kristtst01 commented on src/components/filter-panel.tsx:
I also like it if some prep times are suggested like it was. Natural breakpoints like sub 15m, sub 30m, sub 60m, 60m+

kristtst01 commented on src/components/filter-sidebar.tsx:
This is a good change.

kristtst01 commented on src/components/saved-page-header.tsx:
This is still a problem. Still creating the filter functionality all over again. Just import the component from filter-panel.tsx and use that.

kristtst01 commented on src/components/navbar.tsx:
This is good, but maybe rename the file? I assume this sidebar will be used for the other pages as well?

kristtst01 commented on src/pages/home-page.tsx:
Good changes. This page was a mess because it was used as a mockup first. We definitely need to come up with a better layout for desktop later though. (Not in this issue)

kristtst01 commented on src/pages/recipe-detail.tsx:
Again the ingredients and instructions should be fetched from the mock data. You're already fetching the image and name from there, so just fetch the ingredients and steps from there as well.

kristtst01 commented on src/types/recipe.ts:
Good.

kristtst01 commented on src/utils/recipe-loader.ts:
This is dumb. I mean your solution is good, but fetching from API with category and then just concatenating here is dumb. Don't bother making a change here though. We need to just create an issue for fixing the stupidly split mock data.

kristtst01 reviewed (changes requested):
I've left 9 additional comments.
The page is still broken between desktop and mobile:
<img width="1159" height="1393" alt="image" src="https://github.com/user-attachments/assets/ea0bbc64-1422-48cf-b315-5daaa4a3f12c" />
To reproduce this result, set the width of the page to 775px and swipe the content toward the left. My suggestion for a fix is to just set the breakpoint from mobile design to desktop at a higher width.

PeterBrandal committed:
fixed wraps, now to fix filter

PeterBrandal committed:
fixed wraps, now to fix filter

kristtst01 committed:
fix: add padding to filter and search panel to prevent clipping border around filter option boxes

kristtst01 committed:
fix: filter for Kitchen/country is once again searchable and shows matching countries in scrollable dropdown

kristtst01 committed:
fix: add toggleable prep times for 15m, 30m and 60 under prep time input bar

kristtst01 committed:
fix: update meal-db-helper to get ingredients and instructions from the mock data

kristtst01 committed:
fix: remove hardcoded mock data from recipedetail. Focus first on implementing real data, then we can add more later. Fix: build issues and fix prop mismatch in saved-page.tsc

kristtst01 committed:
refactor: create useCookMode, useRecipeShare and useTriedRecipe hooks. Add toggleInset to eliminate duplicate toggle logic in recipe-detail

kristtst01 committed:
refactor: simplify navbar and remove duplicate
  filter clearing

  - Use navLinks array to eliminate duplicate Link
  components in navbar
  - Remove duplicate clear filters button from
  saved-page (FilterPanel already has one)

kristtst01 committed:
fix: remove fallback for saved-page to show all recipes. Now only show saved recipes

kristtst01 committed:
fix: set overscroll-behaviour to none to prevent screen 'bouncing' when scrolling

kristtst01 committed:
merge branch main into recipe-card-component

kristtst01 reviewed (approved):
I think it's good now. But take a look for yourself. Try to break it.

PeterBrandal commented on src/components/filter-panel.tsx:
Just link the + button in filter to /create-recipe instead of create weekplan.

Keep things to their domain.

PeterBrandal committed:
added a more tap-to-close function for expanded filter (UX)

PeterBrandal committed:
format
